### PR TITLE
Add Troubleshooting doc + instructions to fix mpv crashing on Vivaldi

### LINKED ISSUE #19: mpv does not work by default on Vivaldi >3.0 or when using "proprietary media codecs" (libffmpeg.so)
I'm using the default configuration.
Browser: Vivaldi (I can reproduce it in Chromium)
OS: Arch Linux
I tried all the mpv commands listed [here](https://github.com/alexherbo2/krabby/blob/master/docs/commands.md#mpv) in their corresponding context. mpv doesn't open, and the Shell extension shows an error in vivaldi://extensions.

>Error in event handler: Error: Attempting to use a disconnected port object
>**Context**
>_generated_background_page.html
> **Stack Trace**
> `src/background.js:10 (anonymous function)`
> ```
> const shell = {}
> shell.port = chrome.runtime.connectNative('shell')
> chrome.runtime.onConnectExternal.addListener((port) => {
>   // Send request to the application
>   port.onMessage.addListener((request) => {
>     shell.port.postMessage(request)
>   })
>   // Receive response
>   shell.port.onMessage.addListener((response) => {
>>   port.postMessage(response)   <
>   })
> })
> ```
^The line I enclosed in `>`/`<` is highlighted.
Is this error relevant to mpv not opening or is it just a coincidence? It seemed to appear everytime I tried mpv bindings.

Thank you so much for Krabby! I'm loving the extension so far.

### PR DESCRIPTION
**Pull request by SeerLite**
Closes #19 

I hope the description about the problem isn't too wordy. Feel free to edit out the context of the problem if you think it's unnecessary. Also, the fix depends on `settings['mpv-environment']` being implemented, of course.
I also added an index, to mimic [the example](https://github.com/puppeteer/puppeteer/blob/master/docs/troubleshooting.md) you linked me. I don't know if it looks weird with only one entry though.

I also changed the README and moved the inline krabby logo a little bit lower so that the new link to Troubleshooting didn't wrap around and look weird on desktop. I hope it looks fine like this.

This is my first time ever doing a pull request so forgive me if I forgot anything.

### CODE REVIEW

**Comment by alexherbo2** [comment]
I do not see where `settings['mpv-environment']` is used in Krabby.

Maybe also add an entry in the [Configuration] document, so that variable is documented.

[Configuration]: https://github.com/alexherbo2/krabby/blob/master/docs/configuration.md

**Comment by SeerLite** [comment]
Ah. I misunderstood your last reply in the issue then. I thought adding the actual `mpv-environment` was something you'd do. I hope that wasn't a dumb assumption.

Hey, I can still try to implement it myself. But as I said, I really don't know much javascript.

Should the mpv commands implementations change to something like [this](https://github.com/alexherbo2/krabby/issues/19#issuecomment-619786665) in extension.js then? As I understand, that's the only way to add environment variables to the call of a command. Or is there a way to set them when using `extensions.shell.send()` directly?

**Comment by alexherbo2** [comment]
Add a default `mpv-environment` in [`krabby.js`] and use that variable in the shell call yeah.

[`krabby.js`]: https://github.com/alexherbo2/krabby/blob/master/src/krabby/web/krabby.js

**Comment by SeerLite** [comment]
I mean something like that. Is that alright?

Also, is it ok if instead of using the shell extension directly in the [extension.js mapping](https://github.com/SeerLite/krabby/blob/3f6fbfb392096e5ea84fb8a924b77d8ad3ed1faf/src/krabby/extension/extension.js#L208), it uses `mpvResume`? It seems to accomplish the same thing, the only difference being that `mpvResume` pauses the video too (and I think that's a plus).

**Comment by SeerLite** [comment]
> Also, is it ok if instead of using the shell extension directly in the [extension.js mapping](https://github.com/SeerLite/krabby/blob/3f6fbfb392096e5ea84fb8a924b77d8ad3ed1faf/src/krabby/extension/extension.js#L208), it uses `mpvResume`? It seems to accomplish the same thing, the only difference being that `mpvResume` pauses the video too (and I think that's a plus).

The latest commit is about that. I await your thoughts on this

**Comment by alexherbo2** [comment]
Using the current URL seemed more general purpose when we are not in the video context, because `m` now assumes it will find a video object (pausing and passing the `-start` flag to [mpv]).

[mpv]: https://mpv.io

**Comment by alexherbo2** [comment]
@SeerLite It’s funny that despite you don’t know JavaScript, you managed to make things work.  The changes look quite good to me.

Can you verify one last time everything is working and documented (I see the [Configuration] document has not been updated), and squash the commits into a single one (except the contributing waiver).

[Configuration]: https://github.com/alexherbo2/krabby/blob/master/docs/configuration.md

**Comment by SeerLite** [comment]
> It’s funny that despite you don’t know JavaScript, you managed to make things work. The changes look quite good to me.

Haha, thanks! I did play around a bit with javascript one time and I also do have some experience with other programming languages, so the change here didn't feel too complicated. By doing this I also learned more about git/gihub so that's good too.

> Using the current URL seemed more general purpose when we are not in the video context, because `m` now assumes it will find a video object (pausing and passing the `-start` flag to [mpv](https://mpv.io)).

Oh, I see the problem now. Using `m` on a page without a video will show an error in vivaldi://extensions/. What should I do about it then? Do I change the binding to use a similar implementation instead of `mpvResume()`, or can this be ignored?

**Comment by SeerLite** [comment]
> (I see the [Configuration](https://github.com/alexherbo2/krabby/blob/master/docs/configuration.md) document has not been updated)

I did add an example similar to the one about `mpv-config` in [this commit](https://github.com/alexherbo2/krabby/pull/20/commits/a3ce8c9e48dc0de005ffd49580a4ad382fcf925b) though.

**Comment by SeerLite** [comment]
Alright so I changed `m` to use a similar implementation instead of using `mpvResume()`. I think everything looks good now.

File: docs/troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+1. Add the following to to your configuration located at `~/.config/krabby/config.js`
+   ```javascript
+   const { settings } = krabby
+   
+   // Clear the LD_PRELOAD environment variable. Fixes mpv crash on Vivaldi.
+   settings['mpv-environment'] = {
+     LD_PRELOAD: ''
+   }
+   ```
+1. Run
```
Comment (alexherbo2) [inline docs/troubleshooting.md]: Can you fix the numbering or use an unordered list?

Comment (SeerLite) [reply docs/troubleshooting.md]: Sure! I fixed the ordering.

I commited and rebased to the squashed commit. That's ok, right?
